fix: Refine interactive serve logging and warning messaging (#1483)
SAP/ui5-tooling: internal/documentation/docs/pages/Server.md

## Live Status Banner

When started in an interactive terminal, `ui5 serve` renders a live status banner that shows the server URLs, the root project's name/type/version, the configured UI5 framework, and a single status line that reflects what the server is doing:

- **ready** — the server is idle. Once source files change, a `N projects stale` count appears here until the next request rebuilds them.
- **building** — a build cycle is running; the project counter, current project, and current task update in place.
- **validating cache** / **settling** — the server is checking dependency caches or waiting for a burst of file changes to settle before rebuilding.

> - **error** — the last build failed. The error message is also logged above the banner.

Reviewer: ```suggestion
- **error** — the last build failed. The error message is logged above the banner.
```

> The status line stays pinned at the bottom of the terminal; log messages scroll above it and remain in your terminal scrollback. Log lines that would only duplicate what the banner already shows are suppressed. The banner repaints in place as more information becomes known (project graph resolved, server bound), so early frames may show dim placeholders for sections that have not been populated yet.

Reviewer: ```suggestion
The status line stays pinned at the bottom of the terminal; log messages scroll above it and remain in your terminal scrollback. Log lines duplicating what the banner already shows are suppressed. The banner repaints in place as more information becomes known (project graph resolved, server bound), so early frames may show dim placeholders for sections that have not been populated yet.
```

URL: <url>
```

> The remote-connections warning (when `--accept-remote-connections` is used) is written to `stderr`. Tools and scripts that capture the URL by parsing `ui5 serve` output should run the command in a non-TTY context (for example with `stderr` redirected, or in CI), or set `UI5_CLI_NO_INTERACTIVE=1` to force plain output in an interactive terminal.

Reviewer: ```suggestion
The remote-connections warning (when `--accept-remote-connections` is used) is written to `stderr`. Tools and scripts that capture the URL by parsing `ui5 serve` output should run the command in a non-TTY context (for example, with `stderr` redirected, or in CI), or set `UI5_CLI_NO_INTERACTIVE=1` to force plain output in an interactive terminal.
```
